sql/core/src/test/scala/org/apache/spark/sql/DatasetSuite.scala

```diff
@@ -969,6 +969,20 @@ class DatasetSuite extends QueryTest with SharedSQLContext {
     checkShowString(ds, expected)
   }
 
+  test("SPARK-24442 Show should follow spark.show.default.number.of.rows") {
+    withSQLConf("spark.sql.show.defaultNumRows" -> "100") {
+      (1 to 1000).toDS().as[Int].show
+    }
+  }
+
+  test("SPARK-24442 Show should follow spark.show.default.truncate.characters.per.column") {
+    withSQLConf("spark.sql.show.truncateMaxCharsPerColumn" -> "30") {
+      (1 to 1000).map(x => "123456789_123456789_123456789_")
+        .toDS().as[String]
+        .show
```

Comment:
Plz compare the show result?
https://github.com/apache/spark/blob/9241e1e7e66574cfafa68791771959dfc39c9684/sql/core/src/test/scala/org/apache/spark/sql/UDFSuite.scala#L326